fearless_simd_gen/src/main.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 the Fearless_SIMD Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::{fs::File, io::Write, path::Path};
+
+use clap::{Parser, ValueEnum};
+use proc_macro2::TokenStream;
+
+mod arch;
+mod generic;
+mod mk_neon;
+mod mk_ops;
+mod mk_simd_trait;
+mod mk_simd_types;
+mod ops;
+mod types;
+
+#[derive(Clone, Copy, ValueEnum, Debug)]
+enum Module {
+    SimdTypes,
+    SimdTrait,
+    Ops,
+    Neon,
+}
+
+#[derive(Parser)]
+struct Cli {
+    #[arg(short, long)]
+    module: Option<Module>,
+}
+
+impl Module {
+    fn generate_code(self) -> TokenStream {
+        match self {
+            Module::SimdTypes => mk_simd_types::mk_simd_types(),
+            Module::SimdTrait => mk_simd_trait::mk_simd_trait(),
+            Module::Ops => mk_ops::mk_ops(),
+            Module::Neon => mk_neon::mk_neon_impl(mk_neon::Level::Neon),
+        }
+    }
+
+    fn generate_string(self) -> String {
+        let code = self.generate_code();
+        let parsed = syn::parse_file(&code.to_string()).expect("error parsing {name}");
+        let code_str = prettyplease::unparse(&parsed);
```

Comment:
Would it be simpler/more robust to delegate to the real `rustfmt`? That approach has worked well for us [in ash](https://github.com/ash-rs/ash/blob/55dc692946f52f0dff326bd13b1722d98f52fd2b/generator/src/lib.rs#L3516-L3526), and will likely produce more idiomatic results. That should also enable dropping `syn` entirely for faster iteration.

Reply:
I'm completely open to this, and it sounds like ash is well worth studying as an example.